src/Microsoft.NetCore.Analyzers/Core/Security/ReviewCodeForSqlInjectionVulnerabilities.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    using System.Linq;
+    using System.Collections.Immutable;
+    using Analyzer.Utilities.Extensions;
+    using Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+    using Microsoft.CodeAnalysis.Operations;
+    using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class ReviewCodeForSqlInjectionVulnerabilities : DiagnosticAnalyzer
+    {
+        internal static readonly DiagnosticDescriptor Rule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA3001",
+            nameof(MicrosoftNetCoreSecurityResources.ReviewCodeForSqlInjectionVulnerabilitiesTitle),
+            nameof(MicrosoftNetCoreSecurityResources.ReviewCodeForSqlInjectionVulnerabilitiesMessage),
+            isEnabledByDefault: false,
```

Comment:
My feeling is it depends.  With code bases that have sources and sinks are in the same assembly, then it's useful.  Otherwise with larger code bases where sources and sinks are usually in different assemblies, then this rule probably won't find much.